### PR TITLE
EZP-32239: ERP revision

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EZP-32239
| Versions      | 3.x, master

### CODE REVIEW

File: docs/guide/shop_features/erp.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# ERP [[% include 'snippets/commerce_badge.md' %]]
+
+[[= product_name =]] can be connected to ERP systems. Out of the box it offers connection for the Microsoft Dynamics ERP.
```
Comment (dabrt) [inline docs/guide/shop_features/erp.md:3]:
```suggestion
[[= product_name =]] can be connected to ERP systems. 
Out of the box, it offers connection for the Microsoft Dynamics ERP.
```

File: docs/guide/shop_features/erp.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Existing ERP customers can automatically create an account in the shop without waiting for confirmation from the administrator.
```
Comment (dabrt) [inline docs/guide/shop_features/erp.md:5]:
```suggestion
Existing ERP customers can automatically create an account in the shop without waiting for a confirmation from the administrator.
```

File: docs/guide/shop_features/erp.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+The shop updates customer data from the ERP in real time.
+
+## Stock information
+
+[[= product_name =]] requests real-time stock information from the ERP
```
Comment (dabrt) [inline docs/guide/shop_features/erp.md:10]:
```suggestion
[[= product_name =]] requests real-time stock information from the ERP system
```

File: docs/guide/shop_features/erp.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+and notifies the customer if the stock is lower than the required quantity.
+It is possible to display the real stock as a numeric value as well.
+
+![](img/stock_info_in_basket.png)
+
+## Configuration of price providers
+
+The shop owner can select systems used for calculating prices.
+
+A fallback price provider (e.g. using imported prices) can be configured. It is used if ERP is not available.
```
Comment (dabrt) [inline docs/guide/shop_features/erp.md:20]:
```suggestion
A fallback price provider can be configured (for example, one that uses imported prices). 
It is used if ERP is not available.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/guide/erp_integration/erp_communication.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+|`readdeliveryaddress`|Gets data of a delivery address for the provided party ID|
+|`updatedeliveryaddress`|Updates the ERP data of an existing delivery address|
+|`createdeliveryaddress`|Creates a new delivery address for the provided party ID|
+|`deletedeliveryaddress`|Deletes a delivery address|
+
+You can find the standard messages in `EshopBundle/Resources/config/messages.yml`.
+
+## Additional lines
+
+A response sent to the shop from ERP can contain more information about a product than was requested.
```
Comment (DominikaK) [inline docs/guide/erp_integration/erp_communication.md:30]:
```suggestion
A response that ERP sends to the shop can contain more information about a product than was requested.
```

File: docs/guide/erp_integration/erp_logging.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+All communication (request- and response messages) to and from the ERP is recorded by the `siso_erp.logger`.
+It is a special service of the class `Monolog/Logger`.
+
+To learn more about logging in the shop, see [Logging](../logging/logging.md).
+
+## Accessing logs
+
+Log messages are logged to the database.
+You can use a command to check which messages were exchanged with the ERP system.
```
Comment (DominikaK) [inline docs/guide/erp_integration/erp_logging.md:11]:
```suggestion
You can use a command to check the messages that were exchanged with the ERP system.
```

File: docs/guide/erp_integration/erp_logging.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+`120_complete` indicates the point before specific transport implementation and after all event handlers.
+
+The content includes all changes from the handlers and is a serialized object.
+It contains the state of data which goes either into the mapping or directly into the specific transport (e.g. SOAP).
+The processing time is calculated including the handlers (starting time-stamp before the event).
+
+### `150_mapping`
+
+`150_mapping` indicates the point after serializing the object data into an XML string.
+The mapping step is optional and depends on the concrete implementation of the transport and the configuration of the message.
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/erp_integration/erp_logging.md:52]: "concrete implementation" is a functioning term, see for example https://en.wikipedia.org/wiki/Abstract_factory_pattern

File: docs/guide/erp_integration/remotepriceprovider.md [hunk 8]
```diff
@@ -1,14 +1,18 @@
...
+In that case the shop uses [VatService](../pricing/price_api/localvatservice.md) to get the `vatPercent` by the `vatCode`.
+
 ## Using customer and contact numbers
 
-1. If the customer and contact numbers are set directly in the price request, they are used.
-1. If not, and the customer and contact numbers are set in the BuyerParty, they are used.
-1. If neither customer nor contact number is set and the usage of the template debitor is allowed in the configuration,
-the customer and/or contact number are determined from the [StandardTemplateDebitorService](../pricing/price_api/standardtemplatedebitorservice.md) and used.
+`RemotePriceProvider` uses customer and contact numbers from the price request.
+If they are not set, it uses the numbers set in the BuyerParty.
```
Comment (DominikaK) [inline docs/guide/erp_integration/remotepriceprovider.md:12]:
```suggestion
If they are not set, it uses the numbers set in the Buyer Party.
```

File: docs/guide/erp_integration/remotepriceprovider.md [hunk 9]
```diff
@@ -1,14 +1,18 @@
...
+
+If neither customer nor contact number are set, and template debitor is enabled in the configuration,
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/erp_integration/remotepriceprovider.md:14]: "template debitor", unfortunately, is a domain term, see https://doc.ibexa.co/en/latest/guide/pricing/price_api/standardtemplatedebitorservice

**Review by dabrt — APPROVED** [review]

**Pull request merged**